library/ssl_ciphersuites.c

```diff
@@ -1876,15 +1876,26 @@ int mbedtls_ssl_get_ciphersuite_id( const char *ciphersuite_name )
 
```

Comment:
I'm a bit uncomfortable changing the dependency from `MBEDTLS_CIPHER_C` to `MBEDTLS_CIPHER_C || MBEDTLS_USE_PSA_CRYPTO` before we can test that it works. But I'm not strongly opposed to it either.

Reply:
Also, without further changes, TLS is not going to work well without `cipher.c`, as ciphersuites currently express their dependencies as `MBEDTLS_GCM_C`, so in a cipher-less build, currently, no GCM or CCM ciphersuite would be available. Of course that can and will be fixed in the future, but IMO it shows that now is too early to just declare that `MBEDTLS_SSL_TLS_C` no longer depends on `CIPHER_C` when `USE_PSA` is enabled.

(Also, as I said elsewhere, building without cipher is the kind of thing we've never done, so I expect there will be a number of issues we'll discover while doing it, so that's really an area where we shouldn't declare that something works before we've tested it.)